XR_APILAYER_NOVENDOR_toolkit/layer.cpp

```diff
@@ -617,6 +624,11 @@ namespace {
             // Handle inputs.
             bool requestScreenshot = false;
             if (m_menuHandler) {
+                // NOTE: shouldn't this be relocated at the beginning of updateConfiguration() instead?
```

Comment:
There is no multithreading, so I don't think that is a problem. The options changed in the menu during the call below will only be picked up at the next `xrEndFrame()`. What situation did you have in mind?

Reply:
The sequence is currently the following:

- line 611 calls each post-process handler to prepare for rendering
- line 633 calls the menu which might change some parameters used by the post-processing handlers
- line 695 call each post-process handler to render.

This could break some handlers because the configManager values they are using did change between the call to their update() and the call to their process()/update().

It is not the case yet, but handlers could be implemented in such a way they retrieve config parameters in their rendering phase (case when lazy-initializing textures for example).

I'd suggest to reorder the calls and relocate updateConfiguration() right before m_graphicsDevice->clearRenderTargets();

Reply:
The semantic of `update()` was to make sure that parameters were updated in a single step before any rendering is done. So by this principle, this is not an issue.

I think the only reason the order was like that is because of the `tick()` call in `updateConfiguration()`, which is meant to "count" how many successful `xrEndFrame()` occurred after a config option was changed. Having this after `handleInput()` means we count a successful `xrEndFrame()` in all cases, but I guess that's OK because the delay for committing options is much greater than 1, so being off by 1 is OK.

So feel free to move the call as you suggested.

Reply:
ok